### PR TITLE
✨ Video support for amp-story-360

### LINKED ISSUE #29509: [Story 360] Video support
Video support for `amp-story-360`.

cc @ampproject/wg-stories

### PR DESCRIPTION
**Pull request by processprocess**
Context / Fixes #29509

[demo](https://stories-360.web.app/examples/amp-story-360.amp.html)

### CODE REVIEW

**Review by t0mg — COMMENTED** [review]
Looks great. I could use a few comments as this is getting tricky with all the flags (video/animation/gyro). I wonder if we could abstract that out and have 2 player implementations based on the media type, overriding the play/pause logic and render loop accordingly.

Also the setImage/resize/render calls moving around are having me slightly worried (these are sensitive little methods). I trust that you didn't break anything but maybe now would be a good time to add a couple more tests?

File: extensions/amp-story-360/0.1/amp-story-360.js [hunk 1]
```diff
@@ -660,7 +711,9 @@ export class AmpStory360 extends AMP.BaseElement {
     // the animation (if applicable).
```
[the remark this answers is not in the document]
Comment (processprocess) [reply extensions/amp-story-360/0.1/amp-story-360.js:711]: > Looks great. I could use a few comments as this is getting tricky with all the flags (video/animation/gyro). I wonder if we could abstract that out and have 2 player implementations based on the media type, overriding the play/pause logic and render loop accordingly.
> 
> Also the setImage/resize/render calls moving around are having me slightly worried (these are sensitive little methods). I trust that you didn't break anything but maybe now would be a good time to add a couple more tests?

I agree this branching logic is really hard to follow. 
I made #30371 as an example of the two player implementation.
